SFDScripts/Internal/Hardcore/Hardcore.cs

```diff
@@ -5890,12 +5890,46 @@ public static void ApplyWeaponMod(TPlayer player, int id)
                     }
                 case 6:
                     {
-                        pl.GiveWeaponItem(WeaponItem.BOUNCINGAMMO);
+                        if (player.PrimaryWeapon != null)
+                        {
+                            RifleWeaponItem first = pl.CurrentPrimaryWeapon;
+                            int ammo = (int)Math.Ceiling(((double)first.MaxTotalAmmo / 4));
```

Comment:
Could you refactor this into smaller methods? For example methods called `CalculateBouncyAmmoForPrimaryWeapon` and then `CalculateBouncyAmmoForSecondaryWeapon`